File: Linked-List/1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder res = new StringBuilder();
        for (int i = 0; i < s.length(); ++i) {
            int length = res.length();
            if (length >= 2 && res.charAt(length - 1) == s.charAt(i) && res.charAt(length - 2) == s.charAt(i)) {
                continue;
            }

            res.append(s.charAt(i));
        }
        return res.toString();
    }
}

File: Linked-List/1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string res;
        for (int i = 0; i < size(s); ++i) {
            int length = size(res);
            if (length >= 2 && res[length - 1] == s[i] && res[length - 2] == s[i]) {
                continue;
            }
            res += s[i]; 
        }
        return res;
    }
};
